common/domain/handler_test.go

```diff
@@ -939,3 +939,350 @@ func TestHandler_UpdateIsolationGroups(t *testing.T) {
+						EmitMetric:               true,
+						BadBinaries:              types.BadBinaries{Binaries: map[string]*types.BadBinaryInfo{}},
+						HistoryArchivalStatus:    types.ArchivalStatusEnabled,
+						VisibilityArchivalStatus: types.ArchivalStatusEnabled,
+						HistoryArchivalURI:       "https://test.history",
+						VisibilityArchivalURI:    "https://test.visibility",
+						IsolationGroups:          types.IsolationGroupConfiguration{},
+						AsyncWorkflowConfig:      types.AsyncWorkflowConfiguration{Enabled: false},
+					},
+					ReplicationConfig: &persistence.DomainReplicationConfig{
+						ActiveClusterName: "active",
+						Clusters:          []*persistence.ClusterReplicationConfig{{ClusterName: "active"}, {ClusterName: "standby"}},
+					},
+					ConfigVersion:   1,
+					FailoverVersion: 123,
+					IsGlobalDomain:  false,
+					LastUpdatedTime: time.Now().Add(-24 * time.Hour).UnixNano(),
+				}, nil)
+
+				mockDomainManager.EXPECT().UpdateDomain(gomock.Any(), gomock.Any()).Return(fmt.Errorf("update domain error"))
+			},
+			request: &types.UpdateDomainAsyncWorkflowConfiguratonRequest{
+				Domain: "test-domain",
+				Configuration: &types.AsyncWorkflowConfiguration{
+					Enabled: true,
+				},
+			},
+			expectedErr: fmt.Errorf("update domain error"),
+		},
+		{
+			name:             "global domain update triggers replication",
+			isPrimaryCluster: true,
+			setupMocks: func(mockDomainManager *persistence.MockDomainManager, mockReplicator *MockReplicator) {
+				mockDomainManager.EXPECT().GetMetadata(gomock.Any()).Return(&persistence.GetMetadataResponse{NotificationVersion: 1}, nil)
+				mockDomainManager.EXPECT().GetDomain(gomock.Any(), gomock.Eq(&persistence.GetDomainRequest{Name: "global-test-domain"})).Return(&persistence.GetDomainResponse{
+					Info: &persistence.DomainInfo{Name: "global-test-domain", ID: "domainID"},
+					Config: &persistence.DomainConfig{
+						Retention:                7,
+						EmitMetric:               true,
+						BadBinaries:              types.BadBinaries{Binaries: map[string]*types.BadBinaryInfo{}},
+						HistoryArchivalStatus:    types.ArchivalStatusEnabled,
+						VisibilityArchivalStatus: types.ArchivalStatusEnabled,
+						HistoryArchivalURI:       "https://test.history",
+						VisibilityArchivalURI:    "https://test.visibility",
+						IsolationGroups:          types.IsolationGroupConfiguration{},
+						AsyncWorkflowConfig:      types.AsyncWorkflowConfiguration{Enabled: false},
+					},
+					ReplicationConfig: &persistence.DomainReplicationConfig{
+						ActiveClusterName: "active",
+						Clusters:          []*persistence.ClusterReplicationConfig{{ClusterName: "active"}, {ClusterName: "standby"}},
+					},
+					ConfigVersion:   1,
+					FailoverVersion: 123,
+					IsGlobalDomain:  true,
+					LastUpdatedTime: time.Now().Add(-24 * time.Hour).UnixNano(),
+				}, nil)
+
+				mockDomainManager.EXPECT().UpdateDomain(gomock.Any(), gomock.Any()).Return(nil)
```

Comment:
Sorry to be annoying, but if the test is to really have any serious value at all the updateDomain request should be specified here. Just doing gomock.Any() doesn't really tell you much if anything's broken